File: android_samples_java/19.2 SukritiTest/app/src/main/java/com/sukriti/sukrititest/OneFragment.java
package com.sukriti.sukrititest;

import android.app.Activity;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatImageView;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

public class OneFragment extends Fragment {

    AppCompatImageView incrementImgbtn;
    CallbackInterface callbackInterface;

    @Override
    public void onAttach(@NonNull Activity activity) {
        super.onAttach(activity);
        callbackInterface = (SecondActivity) activity;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
    {
        View view = inflater.inflate(R.layout.fragment_one, container, false);

        incrementImgbtn = view.findViewById(R.id.incrementImgbtn);
        incrementImgbtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                callbackInterface.increment();
            }
        });

        return view;
    }
}

File: android_jetpack/SukritiTest/app/src/main/java/com/sukriti/sukrititest/OneFragment.kt
package com.sukriti.sukrititest

import android.app.Activity
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.widget.AppCompatImageView
import androidx.fragment.app.Fragment
import com.sukriti.sukrititest.databinding.FragmentOneBinding
import com.sukriti.sukrititest.databinding.FragmentThreeBinding

class OneFragment : Fragment() {
    private var _binding: FragmentOneBinding? = null
    private val binding get() = _binding!!
    var callbackInterface: CallbackInterface? = null

    override fun onAttach(activity: Activity) {
        super.onAttach(activity)
        callbackInterface = activity as SecondActivity
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentOneBinding.inflate(inflater, container, false)

        binding.incrementImgbtn!!.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View?) {
                callbackInterface!!.increment()
            }
        })

        return binding.root
    }
}